### DISCUSSION TITLE
Is there any existing method to spawn a vehicle giving a global longitudinal position?

### DISCUSSION THREAD

**Discussion by lev-smolentsev-K1** [metadriverse/metadrive#729] [Q&A] [answered]
For now, I see it by manually iterating by block & each lane calculating the total length and comparing with desired longitude of a vehicle. Am I missing some existing method?

**Comment by QuanyiLi** [comment, marked as answer]
I think it is allowed to generate vehicles by filling in the global position and heading as the argument: https://github.com/metadriverse/metadrive/blob/05ec6c8ea9a9fc77742b8af9967e63b43b4abaae/metadrive/component/vehicle/base_vehicle.py#L121

The reason we calculate the lane length for lanes in each block and place vehicles is that we want to scatter vehicles on the map evenly or randomly and avoid overlapping.

**Reply by lev-smolentsev-K1** [reply]
Thanks for your reply! My reason is that I want to spawn a traffic vehicle in predefined longitude on the predefined map (being a sequence of S or C segments). I achieved that by iterating over each block and checking the length of the road.
